### PR TITLE
feat(QTREES-386): handle outdated nowcast

### PR DESCRIPTION
**Pull request by dnsos**
This PR handles outdated nowcast data. We now only display trees in our color scheme if they have a nowcast value from the current day.

> **Attention**: visually, this is a breaking change because we won't display nowcasts anymore if they are not from the current day.

Screenshot 2023-02-09 at 14 56 52

Screenshot 2023-02-09 at 14 57 00

## What has changed?

We now always query for the nowcast of _today_ and if that is not available we show the outdated trees in grey.

When clicking the question tag next to the grey legend item, a modal opens that explains that some trees are outside the project's scope. It is agreed by the team that we do not mention the fact that outdated nowcast data might be a reason for the grey color.

Edit: updated the description to reflect the agreed-upon UI.

### CODE REVIEW

File: src/components/Modal/Modal.test.tsx
Comment (dnsos) [inline src/components/Modal/Modal.test.tsx:62]: Just noticed that I somehow got distracted and forgot the actual assertion of the disappearing modal.

File: src/layouts/MapLayout.tsx
Comment (vogelino) [inline src/layouts/MapLayout.tsx:113]: We could maybe rename this function to `onOutdatedNowcastChange` because `check` isn't very telling in my opinion. Feel free to ignore though.

**Review by vogelino — APPROVED** [review]
Great job! Very clean and understandable implementation! Love it! :)

I added a small naming comment which you can decide to keep or throw away if you want.

File: src/layouts/MapLayout.tsx
Comment (dnsos) [reply src/layouts/MapLayout.tsx]: Goo idea. This specific line doesn't exist anymore, but I've changed the name of the function in other places.

**Comment by dnsos** [comment]
**Attention**: Do not merge this yet! We need some internal discussion concerning the DB values. @julizet has details.

**Comment by ff6347** [comment]
@dnsos Is this still valid? It needs some conflict fixing

**Comment by dnsos** [comment]
It is an important change, but I'm not sure if we have the go for merging it (after the conflict resolutions). @julizet what do you say?

**Comment by julizet** [comment]
> It is an important change, but I'm not sure if we have the go for merging it (after the conflict resolutions). @julizet what do you say?

I think it's an important feature to know if the nowcast is outdated or not. Like you mentioned @dnsos: it makes the frontend more fail-safe. So we should try to merge this.

However, there ist just one thing we have to keep in mind **before** merging: if the outdated trees are displyed as grey dots how do we display Anlagebäume? 

**Status:**
At our joint meeting with the greenery department, we agreed that the nowcast and the forecast for Anlagenbäume ist not as relevant as for the Straßenbäume (bc they live in better conditions). Plus: the nowcast and forecast for the anlagenbäume might not as accurate as the one for the Straßenbäume since our AI model was trained with sensor data from solely Straßenbäumen. We still want to display the Anlagenbäume though.

**Idea:**
Also display the Anlagenbäume also as grey dots + update the text behind the `?` button in the Legend to (mention that the tree has either outdated values if its a Straßenbaum or unreliable values if its an Anlagenbaum) + add a little tag/icon to each detail view of a tree stating the type (Straßenbaum or Anlagenbaum)

**Comment by dnsos** [comment]
@julizet 

> We still want to display the Anlagenbäume though.

But no nowcast/forecast will be calculated for them, right? If so, I think our modal text still makes sense as-is:

> Bäume, die auf der Karte grau markiert sind, haben entweder noch keinen berechneten Wert für den heutigen Tag oder **sie sind nicht Teil des Vorhersagemodells**.

> + add a little tag/icon to each detail view of a tree stating the type (Straßenbaum or Anlagenbaum)

That's a nice idea! I would suggest to do that in another ticket though. This PR is already quite big.

**Comment by julizet** [comment]
@dnsos 

> But no nowcast/forecast will be calculated for them, right? If so, I think our modal text still makes sense as-is:

Maybe it could be a little bit more explicit: **Straßenbäume**, die auf der Karte grau markiert sind, haben entweder noch keinen berechneten Wert für den heutigen Tag oder sind **Anlagenebäume**  und nicht Teil des Vorhersagemodells.

> That's a nice idea! I would suggest to do that in another ticket though. This PR is already quite big.

Sure! I cerated a ticket for our upcoming Sprint

**Pull request merged**
